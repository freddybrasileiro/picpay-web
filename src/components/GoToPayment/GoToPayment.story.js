import React from "react";
import { storiesOf } from "@storybook/react";
import { ThemeProvider } from "styled-components";
import { IntlProvider } from "react-intl";

import styledComponentsTheme from "styledComponentsTheme";
import { DEFAULT_LOCALE, translationMessages } from "i18n";

import GoToPayment from "components/GoToPayment";

storiesOf("Components/GoToPayment", module)
  .addDecorator(story => (
    <IntlProvider
      locale={DEFAULT_LOCALE}
      messages={translationMessages[DEFAULT_LOCALE]}
    >
      <ThemeProvider theme={styledComponentsTheme}>
        <div
          style={{
            maxWidth: 500,
            margin: "10px 10px",
            backgroundImage: "linear-gradient(264deg,#3c3e64,#1d1e2a)"
          }}
        >
          {story()}
        </div>
      </ThemeProvider>
    </IntlProvider>
  ))
  .add("Default", () => <GoToPayment />);
