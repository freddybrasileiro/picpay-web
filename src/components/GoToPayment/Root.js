import styled from "styled-components";

export default styled.div`
  display: flex;
  text-transform: uppercase;
  align-items: center;
`;
