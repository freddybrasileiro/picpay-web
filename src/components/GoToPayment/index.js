import React from "react";
import PropTypes from "prop-types";
import { Typography, withWidth } from "@material-ui/core";
import { Padding } from "styled-components-spacing";

import Root from "components/GoToPayment/Root";

import DollarIcon from "components/icons/DollarIcon";
import RightArrow from "components/icons/RightArrow";

const GoToPayment = ({ width }) => (
  <Root width={width}>
    <DollarIcon width="16" height="16" />
    <Padding right={1} />
    {width !== "xs" && (
      <React.Fragment>
        <Typography component="p" variant="display2">
          Pagar
        </Typography>
        <Padding right={2} />
      </React.Fragment>
    )}
    <RightArrow width="16" height="16" />
  </Root>
);

const propTypes = {
  width: PropTypes.string
};

const defaultProps = {
  width: "xl"
};

GoToPayment.propTypes = propTypes;
GoToPayment.defaultProps = defaultProps;

export default withWidth()(GoToPayment);
