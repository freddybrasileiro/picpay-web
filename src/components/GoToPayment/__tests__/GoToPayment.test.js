import React from "react";
import { createShallow } from "@material-ui/core/test-utils";

import GoToPayment from "components/GoToPayment";

describe("GoToPayment", () => {
  const shallow = createShallow({ dive: true });

  it("Should render the GoToPayment", () => {
    expect(shallow(<GoToPayment />)).toMatchSnapshot();
  });
});
