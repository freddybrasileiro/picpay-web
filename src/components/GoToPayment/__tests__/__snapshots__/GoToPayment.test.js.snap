// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoToPayment Should render the GoToPayment 1`] = `
<Fragment>
  <GoToPayment
    width="md"
  />
  <EventListener
    onResize={[Function]}
    target="window"
  />
</Fragment>
`;
